### PR TITLE
Add const qualifier for extractColumn result VectorPtr 

### PR DESCRIPTION
**Pull request by pramodsatya**
extractColumn result is never used to allocate result vector, only copies to it.

### CODE REVIEW

**Review by aditi-pandit — APPROVED** [review]

**Review by mbasmanova — APPROVED** [review]

File: velox/exec/RowContainer.h [hunk 1]
```diff
@@ -238,7 +238,7 @@ class RowContainer {
       int32_t numRows,
       RowColumn col,
       vector_size_t resultOffset,
-      VectorPtr& result);
+      const VectorPtr& result);
```
Comment (Yuhta) [inline velox/exec/RowContainer.h:241]: How hard is it to change to `BaseVector&` here?

Comment (aditi-pandit) [reply velox/exec/RowContainer.h:241]: Hi Jimmy, BaseVector could be sufficient here. I presumed we just wanted to keep it safe to ensure that RowContainer always had a valid vector to copy to in these APIs.

@mbasmanova : Is there any reason for shared_ptr usage ?

Comment (mbasmanova) [reply velox/exec/RowContainer.h:241]: BaseVector would be nicer.

Comment (aditi-pandit) [reply velox/exec/RowContainer.h:241]: We need this PR for some changes in WindowFunction use. But @pramodsatya can try what it takes for using BaseVector& in a subsequent PR.

**Pull request closed without merging**